### PR TITLE
Bluetooth: MPL: Add notification to track change

### PR DESCRIPTION
**Pull request by piotrnarajowski**
This adds notifications to do_first_track. Before the change was made, client was never notified about track change when changing group in Media Player.

### CODE REVIEW

**Comment by piotrnarajowski** [comment]
@Thalley This is RFC, could you please take a look?

File: subsys/bluetooth/audio/mpl.c [hunk 1]
```diff
@@ -1052,20 +1052,13 @@ static void do_first_track(struct mpl_mediaplayer *pl)
...
-		/* if we stay at the same track (goto start of track) */
-		set_track_position(0);
-	}
-
 	while (pl->group->track->prev != NULL) {
 		pl->group->track = pl->group->track->prev;
 	}
 
+	media_player.track_pos = 0;
+	do_track_change_notifications(&media_player);
```
Comment (Thalley) [inline subsys/bluetooth/audio/mpl.c:1060]: This change seems to discard 

```c
/* For first track, the position is reset to 0 even */
/* if we stay at the same track (goto start of track) */
set_track_position(0);
```

So it seems that this new proposed change would always send a track change notification, even if the track doesn't change. That doesn't seem right.

As per your commit message, the issue you are trying to fix seems to be with changing groups, and it's unclear to me why `do_first_track` is the right place to do that.

Comment (piotrnarajowski) [reply subsys/bluetooth/audio/mpl.c]: Right... discarded that idea and updated functions related to group change

File: subsys/bluetooth/audio/mpl.c [hunk 2]
```diff
@@ -1162,6 +1162,7 @@ static void do_prev_group(struct mpl_mediaplayer *pl)
 	if (pl->group->prev != NULL) {
 		pl->group = pl->group->prev;
 		do_group_change_notifications(pl);
+		do_track_change_notifications(&media_player);
```
[the remark this answers is not in the document]
Comment (piotrnarajowski) [reply subsys/bluetooth/audio/mpl.c]: Yeah, this makes much more sense now.

File: subsys/bluetooth/audio/mpl.c [hunk 3]
```diff
@@ -1046,14 +1046,22 @@ static void do_next_track(struct mpl_mediaplayer *pl)
 	}
 }
 
-static void do_first_track(struct mpl_mediaplayer *pl)
+static void do_first_track(struct mpl_mediaplayer *pl, bool group_change)
 {
+	bool track_changed;
```
Comment (Thalley) [inline subsys/bluetooth/audio/mpl.c:1051]:
```suggestion
	bool track_changed = false;
```

**Review by Thalley — CHANGES_REQUESTED** [review]
Does this fix the PTS issues?

**Review by Thalley — APPROVED** [review]

File: subsys/bluetooth/audio/mpl.c (hunk 3 above)
Comment (piotrnarajowski) [reply subsys/bluetooth/audio/mpl.c]: Yes, this fixes both PTS issue and [TS Errata](https://bluetooth.atlassian.net/browse/ES-24634)

**Review by larsgk — APPROVED** [review]

**Comment by Thalley** [comment]
@piotrnarajowski does this fix https://github.com/zephyrproject-rtos/zephyr/issues/67312?

**Comment by piotrnarajowski** [comment]
@Thalley Unfortunately no, this is separate issue

**Pull request merged**